src/public/javascripts/departureForm.js

```diff
@@ -0,0 +1,146 @@
+
+const pageForm = document.getElementById('page-form');
+const confirmWarnedDepartureDialog = document.getElementById("confirmWarnedDepartureDialog")
+const continueWithWarnedDate = document.getElementById('continueWithWarnedDate');
+
+const twoHourWarningTexts = Array.from(document.getElementsByClassName("twoHourWarningText"));
+const fortyEightHourWarningTexts = Array.from(document.getElementsByClassName("fortyEightHourWarningText"));
+
+let departureFormSubmitter = undefined;
+
+dialogPolyfill.registerDialog(confirmWarnedDepartureDialog);
+
+const departureDate =  () => {
+  return new Date(
+    Number(departureYear.value),
+    Number(departureMonth.value) - 1,
+    Number(departureDay.value),
+    Number(departureHourTime.value),
+    Number(departureMinuteTime.value)
+  )
+};
+  
+
+function showDepartureDateWarningMessages(providedDate) {
+  console.log(providedDate);
+  twoHourWarningTexts.forEach($element => $element.hidden = isTwoHoursPriorDeparture(providedDate));
+  fortyEightHourWarningTexts.forEach($element => $element.hidden = dateNotMoreThanTwoDaysInFuture(providedDate));
+}
+
+setTimeout(() => {
+    // Instead of dialog confirmination, it will submit the form so post controller validations catches this.
+    if ([
+      departureYear.value, 
+      departureMonth.value,
+      departureDay.value,
+      departureHourTime.value,
+      departureMinuteTime.value
+    ].includes('')) {
+      return undefined;
+    }
+
+  showDepartureDateWarningMessages(departureDate());
+}, 500)
+
+
+pageForm.addEventListener("submit", (e) => {
+    // Instead of dialog confirmination, it will submit the form so post controller validations catches this.
+  if ([
+    departureYear.value, 
+    departureMonth.value,
+    departureDay.value,
+    departureHourTime.value,
+    departureMinuteTime.value
+  ].includes('')) {
+    return undefined;
+  }
+
+  if (departureFormSubmitter) {
+    return undefined;
+  }
+  
+  if (isTwoHoursPriorDeparture(departureDate()) && dateNotMoreThanTwoDaysInFuture(departureDate())) {
+    return undefined;
+  }
+
+  e.preventDefault();
+  departureFormSubmitter = e.submitter;
+  showDepartureDateWarningMessages(departureDate());
+  confirmWarnedDepartureDialog.showModal();
+  
+});
+
+confirmWarnedDepartureDialog.addEventListener("close", (e) => {
+  departureFormSubmitter = undefined;
+});
+continueWithWarnedDate.addEventListener("click", (e) => {
+    pageForm.requestSubmit(departureFormSubmitter);
+});
+
+departureDay.addEventListener("keyup", (e) => {
+  e.target.value = sanitiseValue(e.target.value, 'day');
+  autoTab(departureDay, 'day', departureMonth);
+  showDepartureDateWarningMessages(departureDate());
+});
+
+departureMonth.addEventListener("keyup", (e) => {
+  e.target.value = sanitiseValue(e.target.value, 'month');
+  autoTab(departureMonth, 'month', departureYear);
+  showDepartureDateWarningMessages(departureDate());
+});
```

Comment:
Observation:  I don't think sanitiseValue, sanitiseValue1 and sanitiseValue3 are good functions names.  You use them quite a lot.  I think more descriptive names would work better.

Reply:
Sure, I will edit them and try to make them better.
- They were part of the origin function names.
- Whilst I wanted to edit them at the time, I don't want to go out of the change's scope and add the refactors I wanted to.

Reply:
Maybe a comment about what the regex does in each function would be helpful too.  The function sanitiseValue is easy to understand but sanitiseValue1 and sanitiseValue2 not so much.